includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1232 @@
+	public $button_theme;
+
+	/** @var string configuration option: button shape for the Cash App Pay button. */
+	public $button_shape;
+
+	/**
+	 * Constructs the class.
+	 *
+	 * @since x.x.x
+	 */
+	public function __construct() {
+		parent::__construct(
+			Plugin::CASH_APP_PAY_GATEWAY_ID,
+			wc_square(),
+			array(
+				'method_title'       => __( 'Cash App Pay (Square)', 'woocommerce-square' ),
+				'method_description' => __( 'Allow customers to securely pay with Cash App', 'woocommerce-square' ),
+				'payment_type'       => 'cash_app_pay',
+				'supports'           => array(
+					self::FEATURE_PRODUCTS,
+					self::FEATURE_REFUNDS,
+				),
+				'countries'          => array( 'US' ),
+				'currencies'         => array( 'USD' ),
+			)
+		);
+
+		// Payment method image.
+		$this->icon = $this->get_payment_method_image_url();
+
+		// Transaction URL format.
+		$this->view_transaction_url = $this->get_transaction_url_format();
+
+		// Ajax hooks
+		add_action( 'wc_ajax_square_cash_app_pay_get_payment_request', array( $this, 'ajax_get_payment_request' ) );
+		add_action( 'wc_ajax_square_cash_app_pay_set_continuation_session', array( $this, 'ajax_set_continuation_session' ) );
+		add_action( 'wc_ajax_square_cash_app_log_js_data', array( $this, 'log_js_data' ) );
+
+		// restore refunded Square inventory
+		add_action( 'woocommerce_order_refunded', array( $this, 'restore_refunded_inventory' ), 10, 2 );
+
+		// Admin hooks.
+		add_action( 'admin_notices', array( $this, 'add_admin_notices' ) );
+	}
+
+	/**
+	 * Enqueue the necessary scripts & styles for the gateway.
+	 *
+	 * @since x.x.x
+	 */
+	public function enqueue_scripts() {
+		if ( ! $this->is_configured() ) {
+			return;
+		}
+
+		// Enqueue payment gateway assets.
+		$this->enqueue_gateway_assets();
+	}
+
+	/**
+	 * Payment form on checkout page.
+	 *
+	 * @since x.x.x
+	 */
+	public function payment_fields() {
+		parent::payment_fields();
+		?>
+		<br />
+		<div id="wc-square-cash-app-pay-hidden-fields">
+			<input name="<?php echo 'wc-' . esc_attr( $this->get_id_dasherized() ) . '-payment-nonce'; ?>" id="<?php echo 'wc-' . esc_attr( $this->get_id_dasherized() ) . '-payment-nonce'; ?>" type="hidden" />
+		</div>
+		<form id="wc-square-cash-app-payment-form">
+			<div id="wc-square-cash-app"></div>
+		</form>
+		<?php
+	}
+
+	/**
+	 * Return the gateway-specifics JS script handle.
+	 *
+	 * @since x.x.x
+	 * @return string
+	 */
+	protected function get_gateway_js_handle() {
+		return 'wc-' . $this->get_id_dasherized();
+	}
+
+	/**
+	 * Enqueue the gateway-specific assets if present, including JS, CSS, and
+	 * localized script params
+	 *
+	 * @since x.x.x
+	 */
+	protected function enqueue_gateway_assets() {
+		$is_checkout = is_checkout() || ( function_exists( 'has_block' ) && has_block( 'woocommerce/checkout' ) );
+
+		// bail if not a checkout page or cash app pay is not enabled
+		if ( ! ( $is_checkout || $this->is_configured() ) ) {
```

Comment:
I might be misunderstanding this based on the comment above but I believe this logic is incorrect. 

If `is_checkout` is `false` but `$this->is_configured()` is `true` this will equate to `! ( false || true )` ⇒ `false`, will not return and so as a result it will enqueue the scripts on all pages as long as the gateway is configured. 

Alternatively, if the page is the checkout page and this gateway isn't configured. ie `! ( true || false )` ⇒ `false` and so we'd be loading these scripts if it wasn't configured. 

```suggestion
		if ( ! $is_checkout || ! $this->is_configured() ) {
```

If that's the case, you'll likely need to check if the page is an pay-for-order page too because I suspect you'll need to load these scripts on the pay for order page.

Reply:
Thanks for catching this @james-allan. fixed in https://github.com/woocommerce/woocommerce-square/pull/53/commits/9d3bc1f25a86ccb73e98216aabc09fb13fa9baec